plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerStatsCollector.java

```diff
@@ -0,0 +1,440 @@
+
+        Map<String, String> columnNameToStatisticsName = new HashMap<>();
+        for (String statisticName : singleColumnStatistics) {
+            String columnName = statisticsDao.getSingleColumnStatisticsColumnName(tableObjectId, statisticName);
+            if (columnName == null) {
+                // Table or statistics disappeared
+                continue;
+            }
+
+            if (columnNameToStatisticsName.putIfAbsent(columnName, statisticName) != null) {
+                log.debug("Multiple statistics for %s in %s: %s and %s", columnName, table, columnNameToStatisticsName.get(columnName), statisticName);
+            }
+        }
+        return columnNameToStatisticsName;
+    }
+
+    private TableStatistics readNewTableStatistics(ConnectorSession session, JdbcTableHandle table, SqlServerClient sqlServerClient)
+            throws SQLException
+    {
+        try (Connection connection = connectionFactory.openConnection(session);
+                Handle handle = Jdbi.open(connection)) {
+            RemoteTableName remoteTableName = table.getRequiredNamedRelation().getRemoteTableName();
+            String catalog = remoteTableName.getCatalogName().orElse(null);
+            String schema = remoteTableName.getSchemaName().orElse(null);
+            String tableName = remoteTableName.getTableName();
+
+            NewStatisticsDao statisticsDao = new NewStatisticsDao(handle);
+            Optional<TableLevelInfo> tableLevelInfo = statisticsDao.getTableLevelInfo(catalog, schema, tableName);
+            if (tableLevelInfo.isEmpty() || tableLevelInfo.get().rowCount() == 0L) {
+                return TableStatistics.empty();
+            }
+            long tableObjectId = tableLevelInfo.get().objectId();
+            long rowCount = tableLevelInfo.get().rowCount();
+
+            TableStatistics.Builder tableStatistics = TableStatistics.builder();
+            tableStatistics.setRowCount(Estimate.of(rowCount));
+
+            ImmutableMap<String, JdbcColumnHandle> columns = sqlServerClient.getColumns(session, table)
+                    .stream().collect(toImmutableMap(JdbcColumnHandle::getColumnName, c -> c));
+
+            ConcurrentHashMap<String, ColumnStatsBuilder> columnStats = new ConcurrentHashMap<>();
+            statisticsDao.getColumnStatRanges(tableObjectId).forEach(columnStatsRange -> {
+                JdbcColumnHandle column = columns.get(columnStatsRange.columnName);
+                if (column != null) {
+                    log.info(columnStatsRange.toString());
+                    ColumnStatsBuilder builder = columnStats.computeIfAbsent(columnStatsRange.columnName, ignored ->
+                            new ColumnStatsBuilder(column, rowCount));
+                    builder.add(columnStatsRange);
+                }
+                else {
+                    log.warn("No column for %s: %s", columnStatsRange, columns.keySet());
+                }
```

Comment:
Reverse this condition and do an early return to avoid one level of indentation. This will make it easier to read - do validation first, and then continue with main logic.
```suggestion
                if (column == null) {
                    log.warn("No column for %s: %s", columnStatsRange, columns.keySet());
                    return;
                }
                log.info(columnStatsRange.toString());
                ColumnStatsBuilder builder = columnStats.computeIfAbsent(columnStatsRange.columnName, ignored ->
                            new ColumnStatsBuilder(column, rowCount));
                builder.add(columnStatsRange);
```